stl/inc/format

```diff
@@ -441,20 +441,35 @@ _NODISCARD constexpr const _CharT* _Parse_arg_id(
     // This is where we would parse named arg ids if std::format were to support them.
     _THROW(format_error("Invalid format string."));
 }
-
-_NODISCARD constexpr bool _Is_execution_charset_utf8() {
-#pragma warning(push)
-#pragma warning(disable : 4309) // 'initializing' : truncation of constant value
-#pragma warning(disable : 4566) // character represented by universal-character-name '\u4E00' cannot be represented in
-                                // the current code page
-#pragma warning(disable : 6201) // Index '2' is out of valid index range '0' to '1' for possibly stack allocated buffer
-                                // '_Test_char'
-#pragma warning(disable : 6239) // (<non-zero constant> && <expression>) always evaluates to the result of <expression>.
-                                // Did you intend to use the bitwise-and operator?
-    constexpr char _Test_char[] = "\u4e00";
-    return sizeof(_Test_char) == 4 && _Test_char[0] == '\xe4' && _Test_char[1] == '\xb8' && _Test_char[2] == '\x80';
-#pragma warning(pop)
+#ifdef __clang__ // TRANSITION: LLVM-52549 (clang-cl 14 will define _MSVC_EXECUTION_CHARACTER_SET)
+_NODISCARD constexpr bool _Is_execution_charset_self_synchronizing() {
+    return string_view{__clang_literal_encoding__} == "UTF-8";
+}
+#else // ^^^ clang / msvc vvv
+_NODISCARD constexpr bool _Is_execution_charset_self_synchronizing() {
+
+    // charset codes (as reported by _MSVC_EXECUTION_CHARACTER_SET) that are
+    // self-synchronizing (and "narrow" so 1200 (utf-16) and 1201 (unicodeFFFE) are not listed)
+    // we care about this because if a charset is self-synchronizing then we can search through it
+    // for formatting control characters _without_ generally decoding the format string.
+    switch (_MSVC_EXECUTION_CHARACTER_SET) {

```

Comment:
EDG doesn't define this (as of 17.1 Preview 4):

```
C:\Temp>type meow.cpp
static_assert(_MSVC_EXECUTION_CHARACTER_SET == 65001);

C:\Temp>cl /EHsc /nologo /W4 /std:c++latest /utf-8 /c meow.cpp
meow.cpp

C:\Temp>cl /EHsc /nologo /W4 /std:c++latest /utf-8 /c /BE meow.cpp
meow.cpp
"meow.cpp", line 1: error: identifier "_MSVC_EXECUTION_CHARACTER_SET" is
          undefined
  static_assert(_MSVC_EXECUTION_CHARACTER_SET == 65001);
                ^
```

This isn't a problem for the tests, as we exclude EDG when concepts are required, but actual IntelliSense activates `__cpp_lib_concepts` (by special agreement with the IntelliSense team, as it can survive instantiation errors that would doom `/BE` compilations in our test suite).

I think we need a fallback path for EDG here, and we need to create a work item for them.

Reply:
Filed VSO-1468747 "EDG should implement `_MSVC_EXECUTION_CHARACTER_SET`".